### ISSUE TITLE
Error loop with custom domain validation using external DNS

### ISSUE THREAD

**Issue by RobertBouillon** [aws-amplify/amplify-hosting#4020]
### Environment information

```plain text
Not Applicable / Using AMP1
```

### Steps to reproduce
1. Set up custom domain
2. Start the process / generate the SSL cert
3. Enter the CNAME into the DNS server after 1 day (simulate propagation time)
4. Go back Amplify control panel. DNS authorization failed. Retry button is the only option,
5. Click the retry button
6. A new SSL certificate is generated
7. A new CNAME key is generated
8. The validation fails because Amplify sees the old CNAME in the DNS
9. Go back to step 4.

**Comment by Jay2113** [comment]
Thanks for sharing the additional context. When you click the retry button, it will generate new DNS records (the ACM record might remain the same, but the CloudFront distribution record will change). Therefore, if the domain activation process fails, we recommend deleting the domain and then restarting the process. It is crucial that you add the records to your DNS settings promptly after adding your custom domain in the Amplify console. The AWS Certificate Manager (ACM) immediately starts attempting to verify ownership, and the verification checks become less frequent over time.

**Comment by RobertBouillon** [comment]
If the recommendation is to delete the record, wait for the TTL to expire, and then try again (presumably up to 3 days), that's not clear during the process, and seems less than ideal. 

If the argument is that this is not a defect and is "working as intended," there are a couple problems with that:
1, There is no indication that the DNS record changed when you hit "retry." You have to find out when it fails again.
2. There's no instruction to delete the previous DNS record before retrying

Based on the information I have, I would recommend two possible remedies:
1. Ideally, hitting "retry" will just retry the ACM verification process and not reissue an SSL certificate with new DNS records
2. If "retry" is clicked and the SSL process restarts, it should be clear to the user that the DNS entries previously entered are invalid, and Amplify should provide up to 3 days for the changes to take affect because DNS can take up to 3 days to propagate. The ACM verification process should not fail and stop retrying if the CNAME record exists and is incorrect.

**Comment by frankadrian** [comment]
Im experiencing this exact problem as @RobertBouillon has described it. It's really annoying that you can't just "retry" with the same cloud front domain being used.

**Comment by Jay2113** [comment]
@RobertBouillon @frankadrian Thanks for sharing that feedback. We appreciate you highlighting the confusion around the current `retry` process. We are working on improving that workflow, and in the meantime, we will update our documentation to clarify how it works. Thank you.

**Comment by MagdielCAS** [comment]
I am having the **exact same issue**, and due to another company is managing the DNS for my client I am not able to promptly update the DNS since I need to wait for their answer as I request the DNS changes. Is there another way to Amplify to retry the DNS verification? Its one year since this issue was firstly reported by OP

**Comment by breaker05** [comment]
Any update? Still have this issue. A simple Retry where it does NOT generate a new cloudfront url would be great.

**Comment by mitchellmcneil** [comment]
Same issue here.

**Comment by Lewis15520** [comment]
The exact same issue here. 

Using an external DNS provider that for what ever reason takes 1 hour to propagate CNAME DNS records. By that point Amplify DNS checks have failed and the retry provides a different CloudFront Distrobution URL which will be updated by my DNS provider in another hour. It's a continuious loop.

**Comment by breaker05** [comment]
Yeah this is still an obvious issue and another reason to move away from AWS. They say they can't reproduce but this has been going on for years and shows how lazy this team is.

**Comment by Lewis15520** [comment]
> Yeah this is still an obvious issue and another reason to move away from AWS. They say they can't reproduce but this has been going on for years and shows how lazy this team is.

I've just moved the DNS to Route53, Changed the name servers and waited for everything to propagate. Everything now works flawlessly. But this all could've been avoided.

Amplify just needs a way to set a wait time on when to check for DNS validation after the CloudFront distrobution is created. I don't understand why they wouldn't do that unless Amazon are trying to force DNS to be under Route53.
